pkg/network/dhcpconfigurator/dhcp_configurator.go

```diff
@@ -29,3 +54,26 @@ func (d DhcpConfigurator) ImportConfiguration(ifaceName string) (*cache.DhcpConf
 func (d DhcpConfigurator) ExportConfiguration(config cache.DhcpConfig) error {
 	return d.dhcpConfigurationFactory.CacheDhcpConfigForPid(d.launcherPID).Write(config.Name, &config)
 }
+
+func (d DhcpConfigurator) EnsureDhcp(podInterfaceName string, dhcpConfig cache.DhcpConfig, dhcpOptions *v1.DHCPOptions) error {
+	dhcpStartedFile := fmt.Sprintf("/var/run/kubevirt-private/dhcp_started-%s", podInterfaceName)
+	_, err := os.Stat(dhcpStartedFile)
+	if os.IsNotExist(err) {
+		if err := d.startServer(dhcpConfig, d.advertisingIfaceName, dhcpOptions); err != nil {
+			return fmt.Errorf("failed to start DHCP server for interface %s", podInterfaceName)
+		}
+		newFile, err := os.Create(dhcpStartedFile)
+		if err != nil {
+			return fmt.Errorf("failed to create dhcp started file %s: %s", dhcpStartedFile, err)
+		}
+		newFile.Close()
+	}
+	return nil
+}
+
+func (d DhcpConfigurator) startServer(config cache.DhcpConfig, advertisingIfaceName string, dhcpOptions *v1.DHCPOptions) error {
```

Comment:
As I previously mentioned, I believe `config.IPAMDisabled` should be removed.
The caller of `EnsureDhcp` should avoid calling it if ipam is disabled.
When storing the `dhcpConfig` we can first store a boolean for `DHCPEnabled`. And only if it is `true` store the `dhcpConfig`.

Reply:
> As I previously mentioned, I believe `config.IPAMDisabled` should be removed.

I heard you.

> The caller of `EnsureDhcp` should avoid calling it if ipam is disabled.
> When storing the `dhcpConfig` we can first store a boolean for `DHCPEnabled`. And only if it is `true` store the `dhcpConfig`.

In follow up commits I remove the `dhcpConfig` altogether from the bind mechanism structs that are using it. I see no point to store it as a member only to know if there's a need to start the server. 

Would renaming this `ipamEnabled` to `DHCPEnabled` be a good compromise from your perspective ?